src/test/java/io/cdap/delta/bigquery/BigQueryTargetTest.java

```diff
@@ -41,4 +170,117 @@ public void testStagingBucketName() {
+      BigQueryUtils.getMaximumExistingSequenceNumber(Mockito.anySet(), Mockito.anyString(),
+              Mockito.nullable(String.class), Mockito.any(BigQuery.class),
+              Mockito.nullable(EncryptionConfiguration.class), Mockito.anyInt());
+    }
+  }
+
+  @Test
+  public void testGetMaximumExistingSequenceNumberForNonRetryableFailures() throws Exception {
+    PowerMockito.mockStatic(BigQueryUtils.class, Mockito.CALLS_REAL_METHODS);
+    List<Throwable> exceptions = new ArrayList<>();
+    exceptions.add(new BigQueryException(NOT_IMPLEMENTED_CODE, null));
+    exceptions.add(new RuntimeException());
+    BigQueryTarget bqTarget = new BigQueryTarget(conf);
+    for (Throwable exception: exceptions) {
+      Mockito.when(job.getQueryResults()).thenThrow(exception);
+      try {
+        exceptionRule.expect(RuntimeException.class);
+        bqTarget.initialize(deltaTargetContext);
+      } finally {
+        //Verify no retries
+        PowerMockito.verifyStatic(BigQueryUtils.class, Mockito.times(1));
+        BigQueryUtils.getMaximumExistingSequenceNumber(Mockito.anySet(), Mockito.anyString(),
+                Mockito.nullable(String.class), Mockito.any(BigQuery.class),
+                Mockito.nullable(EncryptionConfiguration.class), Mockito.anyInt());
+      }
+    }
+  }
+
+  @Test
+  public void testCreateConsumerRetryableFailureForGcsCreate() throws Exception {
+    PowerMockito.mockStatic(BucketInfo.class);
+    PowerMockito.doReturn(bucketInfoBuilder).when(BucketInfo.class);
+    BucketInfo.newBuilder(Mockito.nullable(String.class));
+
+    Throwable exception = new StorageException(403, null);
+    Mockito.when(storage.create(Mockito.any(BucketInfo.class))).thenThrow(exception);
+
+    BigQueryTarget bqTarget = new BigQueryTarget(conf);
+    try {
+      //IO Exception thrown is wrapped by Failsafe.Failsafe wraps checked exceptions.
+      exceptionRule.expect(FailsafeException.class);
+      bqTarget.createConsumer(deltaTargetContext);
+    } finally {
+      //Verify at least 1 retry
+      Mockito.verify(storage, Mockito.atLeast(2)).create(Mockito.any(BucketInfo.class));
+    }
+  }
+
+  @Test
+  public void testCreateConsumerRetryableFailureForGcsGet() throws Exception {
+
+    Throwable exception = new StorageException(500, null);
+    Mockito.when(storage.get(Mockito.anyString())).thenThrow(exception);
+
+    BigQueryTarget bqTarget = new BigQueryTarget(conf);
+    try {
+      exceptionRule.expect(exception.getClass());
+      bqTarget.createConsumer(deltaTargetContext);
+    } finally {
+      //Verify at least 1 retry
+      Mockito.verify(storage, Mockito.atLeast(2)).get(Mockito.nullable(String.class));
+    }
+  }
+
+  @Test
+  public void testCreateConsumerNonRetryableFailure() throws Exception {
+    Throwable exception = new StorageException(501, null);
+    Mockito.when(storage.get(Mockito.anyString())).thenThrow(exception);
+
+    BigQueryTarget bqTarget = new BigQueryTarget(conf);
```

Comment:
Create another constructor in BigQueryTarget which takes max retry attempts to avoid blocking the test for minutes